### PR TITLE
Adding Shapes Frontend

### PR DESCRIPTION
**Pull request by MicahElizabeth**
This PR contains classes for Shape, Ellipse, Rectangle, Line, TextShape, Triangle, and ShapesLibraryPlugin. All these classes use [Konva](https://konvajs.github.io/docs/) to display the shapes to the screen. The ShapesLibraryPlugin class is used in src/app/components/common/graphics-window/index.tsx to enable interaction between the engine and front-end.

### CODE REVIEW

File: src/app/prototype.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<html>
```
Comment (allisoninman) [inline src/app/prototype.html]: We should probably remove this file

File: package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
   },
   "devDependencies": {
     "@timkendrick/monaco-editor": "0.0.7",
+    "@types/angular": "^1.6.49",
```
Comment (allisoninman) [inline package.json]: We don't need Angular

File: package.json [hunk 3]
```diff
@@ -29,6 +30,7 @@
     "@types/rimraf": "2.0.2",
     "@types/source-map-support": "0.4.0",
     "@types/webpack": "3.0.12",
+    "angular": "1.5.9",
```
Comment (allisoninman) [inline package.json]: We don't need Angular

File: src/app/prototype.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Observable } from "rxjs";
```
Comment (allisoninman) [inline src/app/prototype.ts]: We should remove this file as well

File: src/app/shapes/ShapesLibraryPlugin.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+    public addTriangle(x1: number, y1: number, x2: number, y2: number, x3: number, y3: number): string {
+        return this.addShape("Triangle", (name) => new Triangle(name, x1, y1, x2, y2, x3, y3));
+    }
+
+    public addLine(x1: number, y1: number, x2: number, y2: number): string {
+        return this.addShape("Line", (name) => new Line(name, x1, y1, x2, y2));
+    }
+    
+    public addImage(): string {
```
Comment (MatthewMeacham) [inline src/app/shapes/ShapesLibraryPlugin.ts]: Actually, @MicahElizabeth should we have just renamed the variable from `imageName` to `_` to make the compiler ignore the error, and still completely implement the interface?

Comment (MicahElizabeth) [reply src/app/shapes/ShapesLibraryPlugin.ts]: I didn't get an error about an incomplete interface, so we're probably okay. I can do that if y'all don't mind doing the approval cycle again.

Comment (OmarTawfik) [reply src/app/shapes/ShapesLibraryPlugin.ts]: You don't need to. This is still implementing the interface. Arguments will be ignored when this function is called.

Comment (MatthewMeacham) [reply src/app/shapes/ShapesLibraryPlugin.ts]: Go ahead and do it, just so someone looking at in the future would know. Sorry I didn't think of it before!

**Comment by OmarTawfik** [comment]
WOHOOO!!!

**Pull request merged**